### PR TITLE
Add react/recommended ESLint rules

### PR DESCRIPTION
**Pull request by pdehaan**
## What does this PR do?

Adds the `react/recommended` ESLint rules:

### Output:

```sh
running bin/pre-commit.js script
executing task "yarn lint"
yarn lint v1.0.0
$ eslint --ext=.js --ext=.json bin/* .

/Users/pdehaan/dev/tmp/del/mozilla-talk/client/coral-admin/src/components/ActionsMenu.js
  24:23  warning  Do not use findDOMNode  react/no-find-dom-node

/Users/pdehaan/dev/tmp/del/mozilla-talk/client/coral-admin/src/components/CommentBox.js
  23:14  warning  Unknown property 'class' found, use 'className' instead  react/no-unknown-property
  24:76  warning  Unknown property 'class' found, use 'className' instead  react/no-unknown-property
  25:18  warning  Unknown property 'class' found, use 'className' instead  react/no-unknown-property
  25:47  warning  Unknown property 'for' found, use 'htmlFor' instead      react/no-unknown-property
  27:14  warning  Unknown property 'class' found, use 'className' instead  react/no-unknown-property
  28:67  warning  Unknown property 'class' found, use 'className' instead  react/no-unknown-property
  29:18  warning  Unknown property 'class' found, use 'className' instead  react/no-unknown-property
  29:47  warning  Unknown property 'for' found, use 'htmlFor' instead      react/no-unknown-property

/Users/pdehaan/dev/tmp/del/mozilla-talk/client/coral-admin/src/components/ui/Header.js
  76:75  warning  Using target="_blank" without rel="noopener noreferrer" is a security risk: see https://mathiasbynens.github.io/rel-noopener  react/jsx-no-target-blank
  81:98  warning  Using target="_blank" without rel="noopener noreferrer" is a security risk: see https://mathiasbynens.github.io/rel-noopener  react/jsx-no-target-blank

/Users/pdehaan/dev/tmp/del/mozilla-talk/client/coral-admin/src/components/UserDetail.js
  192:19  warning  No duplicate props allowed  react/jsx-no-duplicate-props

/Users/pdehaan/dev/tmp/del/mozilla-talk/client/coral-admin/src/routes/Community/components/Table.js
  33:15  warning  No duplicate props allowed  react/jsx-no-duplicate-props
  42:15  warning  No duplicate props allowed  react/jsx-no-duplicate-props

/Users/pdehaan/dev/tmp/del/mozilla-talk/client/coral-framework/components/ClickOutside.js
  25:20  warning  Do not use findDOMNode  react/no-find-dom-node

/Users/pdehaan/dev/tmp/del/mozilla-talk/client/coral-framework/hocs/withCopyToClipboard.js
  9:39  warning  Do not use findDOMNode  react/no-find-dom-node

/Users/pdehaan/dev/tmp/del/mozilla-talk/client/talk-plugin-commentbox/CommentBox.js
  158:9  warning  No duplicate props allowed  react/jsx-no-duplicate-props

✖ 17 problems (0 errors, 17 warnings)
  0 errors, 8 warnings potentially fixable with the `--fix` option.

Done in 11.51s.
```

## How do I test this PR?

`yarn lint`

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,63 +1,74 @@
 {
```
Comment (pdehaan) [inline .eslintrc.json:1]: Sorted `plugins` and `rules` and switched to a consistent 2 space indent.

File: .eslintrc.json [hunk 2]
```diff
@@ -1,63 +1,74 @@
...
+    "promise/no-promise-in-callback": "warn",
+    "promise/no-callback-in-promise": "warn",
+
+    // TODO: Change all these to "error", or remove and use defaults.
+    "react/display-name": "off",
+    "react/jsx-no-duplicate-props": "warn",
+    "react/jsx-no-target-blank": "warn",
+    "react/no-find-dom-node": "warn",
+    "react/no-unknown-property": "warn",
+    "react/prop-types": "off",
```
Comment (pdehaan) [inline .eslintrc.json:33]: Recommend setting this to `"warn"` (or ideally `"error"`), but currently it gives **768 warnings**.

File: .eslintrc.json [hunk 3]
```diff
@@ -1,63 +1,74 @@
...
+  "rules": {
+    "promise/no-return-wrap": "error",
+    "promise/param-names": "error",
+    "promise/catch-or-return": "warn",
+    "promise/no-native": "off",
+    "promise/no-nesting": "warn",
+    "promise/no-promise-in-callback": "warn",
+    "promise/no-callback-in-promise": "warn",
+
+    // TODO: Change all these to "error", or remove and use defaults.
```
Comment (wyattjoh) [inline .eslintrc.json:27]: comments here aren't valid json, should remove it, but the issue is noted :)

Comment (pdehaan) [reply .eslintrc.json:27]: FWIW, comments are valid in ESLint config files (they seem to be using the [**strip-json-comments**](https://github.com/sindresorhus/strip-json-comments) module). Ref: https://eslint.org/docs/user-guide/configuring#comments-in-configuration-files.

**Review by wyattjoh — APPROVED** [review]

**Comment by alinhle** [comment]
👍

**Pull request merged**